### PR TITLE
Room creation transition

### LINKED ISSUE #86: Room-Game Transition
- [x] Create context to be used between component/page loads (#85 )
- [x] Work out room states & state transitions
- [x] Work out commands to send to server

### PR DESCRIPTION
**Pull request by juxd**
Closes #86 

Adds transitions from 'create room' and 'join room' page to 'in-room'.

The 'room' pages are currently at `/room` and `/participant-room` for the host and participants respectively. They will redirect to `/join` if the device is not connected to a room.

### Flow - for host
1. Host enters their name on `/new`
2. Room create attempt will be made, and on success will direct to `/room`, which will render the `HostRoom` component (info to join, start button)
3. On `Cancel`, will leave the room and go back to `/` (home).

### Flow - for user
1. User enters their room and name on `/join` (pending changes from #109 )
2. User joins room on success, and sees status of room based on host room.
3. If host leaves, then an error would be displayed as well as an option to go back to `/`.

### CODE REVIEW

**Comment by juxd** [comment]
@indocomsoft I'm getting a lot of memory leak warnings from React - any idea why?

**Comment by juxd** [comment]
Ignoring the React warning for now. Looks like there is some unsyncable race condition between the Phoenix `onOk` callback and the React unmount handler.

File: Web/src/components/create-room/HostRoom.tsx [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    // Or I am not the host of this room.
+    return <Redirect to="/join" />;
+  }
+
+  return (
+    <CreateRoomContainer>
+      {resultsExist(results) && (
+        <>
+          <h1>Last Game:</h1>
+          {Object.entries(results).map(([guy, score]) => (
```
Comment (indocomsoft) [inline Web/src/components/create-room/HostRoom.tsx]: Did you just assume their genders?

Comment (indocomsoft) [inline Web/src/components/create-room/HostRoom.tsx]:
```suggestion
          {Object.entries(results).map(([name, score]) => (
```

File: Web/src/components/create-room/HostRoom.tsx [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+            <p>{guy}: {score}</p>
```
Comment (indocomsoft) [inline Web/src/components/create-room/HostRoom.tsx]:
```suggestion
            <p>{name}: {score}</p>
```

File: Web/src/components/room/comm/useCommHooks.ts [hunk 3]
```diff
@@ -48,7 +48,7 @@ export default function useCommHooks(
     };
     // Only run once
     // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (indocomsoft) [inline Web/src/components/room/comm/useCommHooks.ts]: Remove this?

**Pull request merged**